[YUNIKORN-992] website update for 1.0.0 release (#158)
apache/yunikorn-site: src/pages/release-announce/1.0.0.md

Apache YuniKorn is a standalone resource scheduler, designed for managing, and scheduling Big Data workloads on container orchestration frameworks like Kubernetes for on-prem and on-cloud use cases.

## Overview
The Apache YuniKorn community has fixed 173 [JIRAs](https://issues.apache.org/jira/issues/?filter=12350818) in this release. 

Release manager: Wilfred Spiegelenburg

Release date: 3 May 2022

## Incompatible changes
As part of this release a number of incompatible changes have been introduced.

### helm upgrade
With the changes made to the helm charts running a helm upgrade from an older version to 1.0.0 might not work.
The workaround is to first uninstall the old version and then perform a clean installation of the current release.

### Configuration changes
Size modifiers are now supported for the queue configuration. 
This changes the way quota values are interpreted in the configuration.
Before 1.0.0 all memory values were interpreted as _megabytes_ and vcore (cpu) values as _milli cores_.
From 1.0.0 onwards memory values are specified in **bytes** and vcores are specified in **cores** [YUNIKORN-629](https://issues.apache.org/jira/browse/YUNIKORN-629).

> To maintain the same interpretation of values in the configuration make the following changes to the pre 1.0.0 configuration:

Reviewer: ```suggestion
To maintain the same interpretation of values in the configuration make the following changes to the pre-1.0.0 configuration:
```

| type   | change | old value    | new value     |
|--------|--------|--------------|---------------|
| memory | add M  | memory: 2048 | memory: 2048M |
| vcore  | add m  | vcore: 250   | vcore: 250m   |
| other  | none   | license: 1   | license: 1    | 

### Resource Display and Communication
The conversion of memory resources of pods and other objects by the k8shim into megabytes could cause a discrepancy between the size accounted for by the Kubernetes tools and YuniKorn.
This could lead to issues around node fits and autoscaling of nodes.
In 1.0.0 all memory resources are converted to **bytes** before being used in calculation and communications [YUNIKORN-1105](https://issues.apache.org/jira/browse/YUNIKORN-1105).

> Resources in all REST response objects are no longer a single string but are JSON structures using a key value pair for each resource type.

Reviewer: ```suggestion
Resources in all REST response objects are no longer a single string but are JSON structures using a key-value pair for each resource type.
```

### Timestamps in REST responses
All timestamps in the rest API are now returned as the number of nanoseconds elapsed since 1 January 1970 UTC.
The time range in a `int64` that can be displayed ranges from the year 1678 or to the year 2262, see [Go Time UnixNano](https://pkg.go.dev/time#Time.UnixNano) for details.
Timestamps in earlier releases were converted to a string as part of the REST response creation with inconsistent layouts [YUNIKORN-1170](https://issues.apache.org/jira/browse/YUNIKORN-1170).

### Removal of old REST API endpoints
The REST API has been updated with endpoints that support specifying partitions. The old versions which do not take a partition argument have been removed:

- `/ws/v1/queues` replaced with `/ws/v1/partition/{partitionName}/queues`
- `/ws/v1/apps` replaced with `ws/v1/partition/{partitionName}/queues/{queueName}/application`
- `/ws/v1/nodes` replaced with `/ws/v1/partition/{partitionName}/nodes`

[YUNIKORN-954](https://issues.apache.org/jira/browse/YUNIKORN-954)

## Highlights

### Plugin mode technical Preview

> As part of this release a new deployment mode is added which is build on top of the Scheduler Plugin Framework.

Reviewer: ```suggestion
As part of this release a new deployment mode is added which is built on top of the [Scheduling Framework](https://kubernetes.io/docs/concepts/scheduling-eviction/scheduling-framework/).
```

> This is a major change in the way the scheduler k8shim works. As the implementation is not considered mature as yet the deployment mode is considered a tech preview.

Reviewer: ```suggestion
This is a major change in the way the scheduler k8shim works. As the implementation is not considered mature yet the deployment mode is considered a tech preview.
```
